modules/transport/Transport.js

```diff
@@ -0,0 +1,265 @@
+import {
+    MESSAGE_TYPE_EVENT,
+    MESSAGE_TYPE_RESPONSE,
+    MESSAGE_TYPE_REQUEST
+} from './constants';
+
+/**
+ * Stores the currnet transport backend that have to be used. Also implements
+ * request/response mechanism.
+ */
+export default class Transport {
```

Comment:
why doesn't this class extend EventEmitter? if would simplify the implementation quite a bit, wouldn't it?